### ISSUE TITLE
p5.Color vs number[]

### ISSUE THREAD

**Issue by lorenzo-lipparini** [lorenzo-lipparini/CNR-project#4]
In the commit 43770c510924489cda723db2dba9639514d61665, all `p5.Vector` and `p5.Color` objects were replaced with number arrays to make them easier to animate.

As it turns out, this resulted in a `+10s` rendering time in the **increasing-iterations** animation in **menger-sponge** (from `69s` to `79s`).

Trying to isolate the problem, the `ambientMaterial()` function call was removed in the `Cube.draw()` method, and the time taken to render the animation then seemed to be the same as that taken when running the previous version of the program (`66s` in both cases).

This suggests that the loss in performance boils down to the way color is passed as a parameter to the `ambientMaterial()` function, and looking at the implementation in the p5 library this actually makes sense, since when the function is called its arguments are parsed using the `color()` function which behaves the following way:

- If the first argument is of type `p5.Color`, just return that
- In any other case, pass the parameters to the `p5.Color` costructor and return the result

This implies a lot more work when the color is given as an array, since all the possible input types and color modes have to be checked, resulting in the loss of time that had been observed.

In conclusion, when working with really complex animations, it might be the case to stick with the older approach.

**Comment by lorenzo-lipparini** [comment]
With all the improvements made to the animation library, going back to `p5.Color` would give a negligible gain in performance compared to the loss in readability, hence the decision to close this issue.
